backend/src/event/Event.ts

```diff
@@ -246,55 +352,68 @@ export class Event {
     return this
   }
 
-  setByBasicTx(
+  setByBasicTx(userId: number, transactionId: number, amount: decimal): Event {
+    this.setByBasicUser(userId)
+    this.transactionId = transactionId
+    this.amount = amount
+
+    return this
+  }
+
+  setByBasicTxX(
     userId: number,
-    xUserId?: number,
-    xCommunityId?: number,
-    transactionId?: number,
-    amount?: decimal,
+    transactionId: number,
+    amount: decimal,
+    xUserId: number,
+    xCommunityId: number,
   ): Event {
-    this.setByBasicUser(userId)
-    if (xUserId) this.xUserId = xUserId
-    if (xCommunityId) this.xCommunityId = xCommunityId
-    if (transactionId) this.transactionId = transactionId
-    if (amount) this.amount = amount
+    this.setByBasicTx(userId, transactionId, amount)
+    this.xUserId = xUserId
+    this.xCommunityId = xCommunityId
 
     return this
   }
 
-  setByBasicCt(userId: number, contributionId: number, amount?: decimal): Event {
+  setByBasicCt(userId: number, contributionId: number, amount: decimal): Event {
     this.setByBasicUser(userId)
-    if (contributionId) this.contributionId = contributionId
-    if (amount) this.amount = amount
+    this.contributionId = contributionId
+    this.amount = amount
 
     return this
   }
 
-  setByBasicRedeem(userId: number, transactionId?: number, contributionId?: number): Event {
-    this.setByBasicUser(userId)
-    if (transactionId) this.transactionId = transactionId
-    if (contributionId) this.contributionId = contributionId
+  setByBasicCtMsg(
+    userId: number,
+    contributionId: number,
+    amount: decimal,
+    messageId: number,
+  ): Event {
+    this.setByBasicCt(userId, contributionId, amount)
+    if (messageId) this.messageId = messageId
```

Comment:
the "if(messageId)" can be removed, because the messageId-parameter is mandatory.

Reply:
missed it, solved